03-path-application-development/309-deploying-a-chart-repository/readme.adoc

```diff
@@ -0,0 +1,264 @@
+access to the bucket, to do so use:
+
+    additionalPolicies:
+      node: |
+        [
+          {
+            "Sid": "AllowListObjects",
+            "Effect": "Allow",
+            "Action": [
+              "s3:ListBucket"
+            ],
+            "Resource": "arn:aws:s3:::{BUCKET_NAME}"
+          },
+          {
+            "Sid": "AllowObjectsCRUD",
+            "Effect": "Allow",
+            "Action": [
+              "s3:DeleteObject",
+              "s3:GetObject",
+              "s3:PutObject"
+            ],
+            "Resource": "arn:aws:s3:::{BUCKET_NAME}/*"
+          }
+        ]
+
+WARNING: Make sure to replace `{BUCKET_NAME}` in each of the statements with the
+`$CHART_BUCKET` without `s3://`
+
+After you have saved the file you can update the cluster thus adding the new
+policies to the nodes.
+
+    kops update cluster example.cluster.k8s.local --yes
+
+To test this we need to first remove the deployed `chartmuseum`.
+
+    helm delete chartmuseum --purge
+
+Then we can add the values on the `helm install` cli like so.
+
+    helm install incubator/chartmuseum \
+      --set env.open.STORAGE=amazon \
+      --set env.open.STORAGE_AMAZON_BUCKET={BUCKET_NAME} \
+      --set env.open.STORAGE_AMAZON_PREFIX="" \
+      --set env.open.STORAGE_AMAZON_REGION=eu-west-1 \
+      --set env.open.DISABLE_API=false \
+      --name chartmuseum
+
+This will deploy a cluster but it won't be publicly accessible. You will still
+need to `kubectl port-forward` to access the repo.
+
+    kubectl port-forward $(kubectl get po -l app=chartmuseum -o jsonpath="{.items..metadata.name}") 8080:8080
+
+With port `8080` forwarded you can now open your browser and view the landing
+page.
+
+    # on macOS
+    open http://localhost:8080
+
+Now that you have access to the server you can add the repo to your local `helm`
+client so that you can deploy a repo into it.
+
+    helm repo add chartmuseum http://localhost:8080
+
+Then we can test packaging the sample application.
+
+    $ cd sample/
+    $ helm package .
+    Successfully packaged chart and saved it to: /03-path-application-development/309-deploying-a-chart-repository/sample/sample-1.0.0.tgz
+
+With this packaged chart we then can push it to the port-forwarded repo.
+
+    $ curl --data-binary "@sample-1.0.0.tgz" http://localhost:8080/api/charts
+    {"saved":true}
+
+To verify the package was added properly you can `curl` the list endpoint.
+
+    $ curl http://localhost:8080/api/charts
+    {"sample":[{"name":"sample","home":"https://github.com/arun-gupta/kubernetes-aws-workshop","sources":["https://github.com/arun-gupta/kubernetes-aws-workshop"],"version":"1.0.0","description":"My first Haml chart","keywords":["java","javaee","mysql","wildfly","wildfly swarm"],"maintainers":[{"name":"Arun Gupta","email":"arun.gupta@gmail.com"}],"urls":["charts/sample-1.0.0.tgz"],"created":"2018-03-09T17:55:43Z","digest":"f01fb09ecb0e0f8f9789aff8343feebab835986d440127471762cfcc5ec2cc24"}]}
+
```

Comment:
Explain what is this output? It seems to be referring to an older version of the workshop too.

Reply:
Yeah, I will do that in the docs FWIW, I'm using the 306 sample chart and just having them package it. Would it make sense for me to update those urls?